localstack/services/kms/provider.py

```diff
@@ -245,131 +416,139 @@ def list_retirable_grants(
-
-        kwargs = {}
-        if signing_algorithm.startswith("RSA"):
-            if "PKCS" in signing_algorithm:
-                kwargs["padding"] = padding.PKCS1v15()
-            elif "PSS" in signing_algorithm:
-                kwargs["padding"] = padding.PSS(
-                    mgf=padding.MGF1(hashes.SHA256()), salt_length=padding.PSS.MAX_LENGTH
-                )
-            else:
-                LOG.warning("Unsupported padding in SigningAlgorithm '%s'", signing_algorithm)
-
-        if "SHA_256" in signing_algorithm:
-            kwargs["algorithm"] = hashes.SHA256()
-        elif "SHA_384" in signing_algorithm:
-            kwargs["algorithm"] = hashes.SHA384()
-        elif "SHA_512" in signing_algorithm:
-            kwargs["algorithm"] = hashes.SHA512()
-        else:
-            LOG.warning("Unsupported hash type in SigningAlgorithm '%s'", signing_algorithm)
-        if signing_algorithm.startswith("ECDSA"):
-            kwargs["signature_algorithm"] = ec.ECDSA(algorithm=kwargs.pop("algorithm", None))
+    # We currently act on neither on KeySpec setting (which is different from and holds values different then
+    # KeySpec for CreateKey) nor on NumberOfBytes. Instead, we generate a key with a key length that is "standard" in
+    # LocalStack.
+    #
+    # TODO We also do not use the encryption context. Should reuse the way we do it in encrypt / decrypt.
+    def _generate_data_key(self, key_id: str, context: RequestContext):
+        key = self._get_store(context).get_key(key_id)
+        # TODO Should also have a validation for the key being a symmetric one.
+        self._validate_key_for_encryption_decryption(key)
+        crypto_key = KmsCryptoKey("SYMMETRIC_DEFAULT")
+        return {
+            "KeyId": key_id,
+            "Plaintext": crypto_key.key_material,
+            "CiphertextBlob": key.encrypt(crypto_key.key_material),
+        }
+
+    @handler("GenerateDataKey", expand=False)
+    def generate_data_key(
+        self, context: RequestContext, request: GenerateDataKeyRequest
+    ) -> GenerateDataKeyResponse:
+        result = self._generate_data_key(request.get("KeyId"), context)
+        return GenerateDataKeyResponse(**result)
+
+    @handler("GenerateDataKeyWithoutPlaintext", expand=False)
+    def generate_data_key_without_plaintext(
+        self, context: RequestContext, request: GenerateDataKeyWithoutPlaintextRequest
+    ) -> GenerateDataKeyWithoutPlaintextResponse:
+        result = self._generate_data_key(request.get("KeyId"), context)
+        result.pop("Plaintext")
+        return GenerateDataKeyWithoutPlaintextResponse(**result)
 
-        # generate signature
-        signature = key_pair["_key_"].sign(data=message, **kwargs)
+    @handler("Sign", expand=False)
+    def sign(self, context: RequestContext, request: SignRequest) -> SignResponse:
+        key = self._get_store(context).get_key(request.get("KeyId"))
```

Comment:
nit: `self._get_store(context).get_key(key_id)` seems to be a pretty common pattern in here. We could add a small util function `self._lookup_key(context, key_id)` that encapsulates and performs the lookup.

Reply:
Done.